### ISSUE TITLE
Docker deployment questions (SSL/Email)

### ISSUE THREAD

**Issue by btbjosh** [CachetHQ/Docker#171]
The Docker deployment was the only deployment of Cachet I could actually get working (as i'm not a php developer). I'm also a little unfamiliar with Docker, except for basic commands. The current documentation assumes that you know what to do when you need to reboot and how to get ssl working inside your docker container. Are there any resources anyone could point me to in order to properly get this ssl working (preferably with the letsencrypt container of nginx)?

Also, when I went through the `/setup` process, and got it going, the `.env` file is not configured with the Mail settings that I specified during setup so none of that works.

If I manually change the `.env` file in my docker container, does that mean I will have to do that each time my container is launched anew (ie. on reboot of my digital ocean droplet)?

Thanks for any help with this.

**Comment by btbjosh** [comment]
Any answer as to why this is the case?

> Also, when I went through the /setup process, the .env file is not configured with the Mail settings that I specified during setup so none of that works.

**Comment by djdefi** [comment]
Hi @btbjosh - For this Docker image, all of the settings are really meant to be populated via [runtime Docker ENV variables](https://docs.docker.com/engine/reference/run/#env-environment-variables). Editing the `.env` within the container itself will not persist through restarts of the container.

You can see all of the variables that we support currently listed inside of https://github.com/CachetHQ/Docker/blob/master/conf/.env.docker

I recommend adding any variables that you need to set (including mail settings) to the https://github.com/CachetHQ/Docker/blob/master/docker-compose.yml `environment:` section. 

#158 in on the radar to help with SSL / letsencrypt support overall

#160 has some tips on how to get a reverse proxy such as nginx working, however it requires some modification to the Cachet source code that we do not yet have configurable via `ENV` variable.

**Comment by btbjosh** [comment]
@djdefi , Thanks for the tips! I actually already tried adding my `environment` settings in my _docker-compose.yml_ file... For some reason, when I ran `docker exec -i cachetdocker_cachet_1 cat .env` after a new `docker-compose up` (after modifying my _docker-compose.yml_ environment settings), it still showed the default settings.

**Comment by djdefi** [comment]
Ah yeah there are some issues with the startup which should be fixed as of #166 

We don't have a image published to docker hub with those fixes yet, but if you want to try cachet 2.3.10 out with the latest  Docker image fixes, you should be able checkout this repo's `master`branch, and then edit the `cachet` service section in `docker-compose.yml` to look like the following: 

```
  cachet:
    build:
      context: .
      args:
        cachet_ver: v2.3.10
    ports:
      - 80:80

...
```

**Comment by djdefi** [comment]
We haven't heard back from you, I wanted to check in one last time. If you need anything else at all, just let us know.

For now I'll go ahead and mark this as closed.
